uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -227,14 +227,12 @@ define([
         },
 
         /**
-         * Delete a leave request
+         * Deletes the leave request
          *
-         * @return {Promise} Resolved with {Object} Deleted Leave request
+         * @return {Promise}
          */
         delete: function () {
-          this.is_deleted = true;
-
-          return LeaveRequestAPI.update(this.toAPI());
+          return LeaveRequestAPI.delete(this.id);
```

Comment:
Previously we had `is_deleted` key to know if a leave request is deleted, how do we know the same now? Because when we delete a leave request, front end also need to remove the same from the list. Right now the only option for that seems to be refreshing the whole list.But that does not sound very efficient. 

Also when the delete method is called the record is deleted from Backend, but still we are keeping the object for the same instance in the frontend. This again does not look right.

I do not have a proper answer to this, please let me know what plan do you have in mind to deal with this situation.